policy-controller/src/lib.rs

```diff
@@ -3,25 +3,298 @@
 mod admission;
 pub mod index_list;
 pub use self::admission::Admission;
-use anyhow::Result;
+pub use linkerd_policy_controller_core::IpNet;
+pub use linkerd_policy_controller_grpc as grpc;
+pub use linkerd_policy_controller_k8s_api as k8s;
+pub use linkerd_policy_controller_k8s_index::{inbound, outbound, ClusterInfo, DefaultPolicy};
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use futures::prelude::*;
+use index_list::IndexList;
+use k8s::{api::apps::v1::Deployment, Client, ObjectMeta, Resource};
+use k8s_openapi::api::coordination::v1 as coordv1;
+use kube::{api::PatchParams, runtime::watcher};
+use kubert::LeaseManager;
 use linkerd_policy_controller_core::inbound::{
     DiscoverInboundServer, InboundServer, InboundServerStream,
 };
 use linkerd_policy_controller_core::outbound::{
     DiscoverOutboundPolicy, OutboundDiscoverTarget, OutboundPolicy, OutboundPolicyStream,
 };
-pub use linkerd_policy_controller_core::IpNet;
-pub use linkerd_policy_controller_grpc as grpc;
-pub use linkerd_policy_controller_k8s_api as k8s;
-pub use linkerd_policy_controller_k8s_index::{inbound, outbound, ClusterInfo, DefaultPolicy};
-use std::{net::IpAddr, num::NonZeroU16};
+use linkerd_policy_controller_k8s_index::ports::parse_portset;
+use linkerd_policy_controller_k8s_status::{self as status};
+use std::{net::IpAddr, net::SocketAddr, num::NonZeroU16, sync::Arc};
+use tokio::{sync::mpsc, time::Duration};
+use tonic::transport::Server;
+use tracing::{info, info_span, instrument, Instrument};
+
+const DETECT_TIMEOUT: Duration = Duration::from_secs(10);
+const LEASE_DURATION: Duration = Duration::from_secs(30);
+const LEASE_NAME: &str = "policy-controller-write";
+const RENEW_GRACE_PERIOD: Duration = Duration::from_secs(1);
+
+#[derive(Debug, Parser)]
+#[clap(name = "policy", about = "Linkerd 2 policy controller")]
+pub struct Args {
```

Comment:
It's probably only useful for this to be _pub_ if the individual arguments/subtypes are as well.